src/main/java/by/itacademy/javaenterprise/knyazev/dao/AbstractDAO.java

```diff
@@ -0,0 +1,18 @@
+package by.itacademy.javaenterprise.knyazev.dao;
+
+import javax.persistence.EntityManager;
+
+public abstract class AbstractDAO<T> implements DAO<T> {
+	
+	protected EntityManager entityManager;
```

Comment:
```suggestion
	private final EntityManager entityManager;
```

- [ ] add `protected EntityManager getEntityManager()`